Update the unit test following changes in injectCredentialsAndInvoke

hex-y opened the pull request:
*Issue #18*

*Description of changes:*
cloudformation-cli-java-plugin made a change to overload injectCredentialsAndInvoke, this breaks our unit test. The change is to fix the build.
https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/pull/338

Update the pom.xml for the repository url, as http was no longer supported.

Update the aws-cloudformation-rpdk-java-plugin version to [2.0.0, 3.0.0)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

siyuxingaws commented on aws-rds-dbproxy/pom.xml:
```diff
@@ -30,7 +30,7 @@
             <groupId>software.amazon.cloudformation</groupId>
             <artifactId>aws-cloudformation-rpdk-java-plugin</artifactId>
-            <version>2.0.0</version>
+            <version>[2.0.0, 3.0.0)</version>
```
Braces is mismarch, should be ``[2.0.0, 3.0.0]``

nmmolletaws commented:
LGTM. Two q - 1) what's the warning we're suppressing 2) PR breaks import order, please reformat.

hex-y commented:
updated the commit:

> 1) what's the warning we're suppressing 

We are suppressing this warning in unit test:
```
[WARNING] /Users/hexiny/resource-providers-rds-proxy/aws-rds-dbproxytargetgroup/src/test/java/software/amazon/rds/dbproxytargetgroup/CreateHandlerTest.java:[98,44] unchecked method invocation: method injectCredentialsAndInvoke in class software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy is applied to given types
  required: RequestT,java.util.function.Function<RequestT,ResultT>
  found: com.amazonaws.services.rds.model.DescribeDBProxyTargetGroupsRequest,java.util.function.Function
```


> 2) PR breaks import order, please reformat.

updated
